### PR TITLE
Try1.7.6

### PR DESCRIPTION
**Pull request by cbouss**
Changes:
- use source distribution instead of git checkout
- remove r outputs (currently built by aggregateR feedstock)
- build libxgboost in libxboost output (instead of global output, then file copies)
- do not use bash on windows
- refresh unvendor libxgboost from py-xgboost patch

https://github.com/dmlc/xgboost/tree/v1.7.6
https://github.com/dmlc/xgboost/blob/v1.7.6/CMakeLists.txt
https://github.com/dmlc/xgboost/blob/v1.7.6/python-package/setup.py

### CODE REVIEW

**Review by JeanChristopheMorinPerso — APPROVED** [review]
Looks good to me, but I didn't analyze the generated libraries.

**Comment by cbouss** [comment]
> Looks good to me, but I didn't analyze the generated libraries.

I extracted the packages and looked at the logs. The content looks good to me.
Only py-xgboost uses libxgboost on our main channel, and the coupling is tight, so I see it as low risks.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,73 +1,51 @@
...
-    - python
-    - setuptools
-    - pip
-    - wheel
 
 outputs:
   - name: libxgboost
     script: install-libxgboost.sh  # [not win]
-    script: install-win-wrapper.bat  # [win]
+    script: install-libxgboost.bat  # [win]
```
Comment (danpetry) [inline recipe/meta.yaml]: can these be called `build-libxgboost` etc now?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,73 +1,51 @@
...
         - llvm-openmp  # [osx]
       run:
         - llvm-openmp  # [osx]
         - _openmp_mutex  # [linux]
+    test:
+      commands:
+        - test -f ${PREFIX}/lib/libxgboost${SHLIB_EXT}       # [unix]
+        - if not exist %LIBRARY_BIN%\\xgboost.dll exit 1     # [win]
 
   - name: _py-xgboost-mutex
```
Comment (danpetry) [inline recipe/meta.yaml:50]: what does this mutex do? afaiu it's not like conda will install two different versions of py-xgboost anyway?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -117,7 +92,6 @@ outputs:
     requirements:
```
Comment (danpetry) [inline recipe/meta.yaml:92]: probably this could also be handled better, although maybe better to do this when there's a -gpu variant

**Review by danpetry — COMMENTED** [review]
Thanks, much cleaner

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [reply recipe/meta.yaml:50]: It doesn't do much, but I am inclined to revisit the day we had a gpu variant.

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml]: addressed in new commit

**Review by danpetry — APPROVED** [review]

**Pull request merged**